### PR TITLE
[sdk-manage] Generate sailfishos-bundle.json. Fixes JB#54178

### PR DESCRIPTION
**Pull request by vigejolla**

### CODE REVIEW

File: sdk-setup/src/sdk-manage [hunk 1]
```diff
@@ -1526,6 +1527,35 @@ amend_target_rpm_macros() {
...
+    "implicitImports": [
+        "__javascriptQt5__"],
+    "supportedImports": [
+EOF
+    IFS=$'\n'; imports=($(sed -n -e 's,^\s*exports: \["\([^/]\+\)/[^ ]\+ \([0-9.]\+\)"\],"\1 \2",p' \
+        $libdir/qt5/qml/**/plugins.qmltypes \
+        |sort -u))
+    printf '        %s,\n' "${imports[@]}"|sed '$s/,$//' >> $bundlefile
+    cat - >> $bundlefile <<EOF
+    ]
```
[the remark this answers is not in the document]
Comment (vigejolla) [reply sdk-setup/src/sdk-manage:1554]: Annoying - why didn't I think of that :-D

[the remark this answers is not in the document]
Comment (vigejolla) [reply sdk-setup/src/sdk-manage]: done

File: sdk-setup/src/sdk-manage [hunk 2]
```diff
@@ -1754,6 +1784,8 @@ install_target() {
 
     rebuild_rpmdb "target:$target" || return
 
+    generate_bundle "$target"
```
[the remark this answers is not in the document]
Comment (vigejolla) [reply sdk-setup/src/sdk-manage]: moved the call to synchronize_target

File: sdk-setup/src/sdk-manage [hunk 3]
```diff
@@ -1526,6 +1527,35 @@ amend_target_rpm_macros() {
 END
 }
 
+#generates sailfishos-bundle.json for QtCreator
+generate_bundle() {
```
[the remark this answers is not in the document]
Comment (vigejolla) [reply sdk-setup/src/sdk-manage]: renamed

**Review by martyone — APPROVED** [review]
Two nitpicks: Just a bit above "END" is used instead of "EOF", `$bundlefile` as a target of redirection operator should be quoted.

**Comment by vigejolla** [comment]
> Two nitpicks: Just a bit above "END" is used instead of "EOF", `$bundlefile` as a target of redirection operator should be quoted.

```
$ grep -c END sdk-manage
2
$ grep -c EOF sdk-manage
11
```

Sure, I can change those two ENDs to EOFs as well if you want?

**Pull request merged**